ui/spinner.js

```diff
@@ -203,20 +208,31 @@ return $.widget( "ui.spinner", {
 
 	_draw: function() {
 		var uiSpinner = this.uiSpinner = this.element
-			.addClass( "ui-spinner-input" )
 			.attr( "autocomplete", "off" )
 			.wrap( this._uiSpinnerHtml() )
 			.parent()
 				// add buttons
 				.append( this._buttonHtml() );
 
+		this._addClass( this.uiSpinner, "ui-spinner", "ui-widget ui-widget-content" );
+		this._addClass( "ui-spinner-input" );
+
 		this.element.attr( "role", "spinbutton" );
 
 		// button bindings
-		this.buttons = uiSpinner.find( ".ui-spinner-button" )
+		this.buttons = uiSpinner.find( "a" )
 			.attr( "tabIndex", -1 )
-			.button()
-			.removeClass( "ui-corner-all" );
+			.button();
+
+			// Right now button does not support classes once it does adjust this with classes
+		this._removeClass( this.buttons, "ui-corner-all" );
+
+		this._addClass( this.buttons.eq( 0 ), "ui-spinner-button ui-spinner-up" );
+		this._addClass( this.buttons.eq( 1 ), "ui-spinner-button ui-spinner-down" );
+		this._addClass( this.buttons.first().find( "span span" ), null,
```

Comment:
Could replace the null with "ui-spinner-button ui-spinner-up" and drop the call above, same for the other button. Doesn't make sense to mix `.eq()` and `.first/last()`, too.